pkg/agent/variables.go

```diff
@@ -171,10 +171,21 @@ func isVHD(profile *datamodel.AgentPoolProfile) string {
 }
 
 func getOutBoundCmd(nbc *datamodel.NodeBootstrappingConfiguration, cloudSpecConfig *datamodel.AzureEnvironmentSpecConfig) string {
+	if nbc.ContainerService == nil {
+		return "nbc.ContainerService is nil"
```

Comment:
hrm...we may need a simpler way to fix these, we can't return error strings in any of these cases as-is

Reply:
Same with the comment above, any object with the potential to be nil needed to be checked. Same suggestion of returning an empty string?